docs/source/library-user-guide/working-with-exprs.md

```diff
@@ -50,6 +50,29 @@ As another example, the SQL expression `a + b * c` would be represented as an `E
 
 As the writer of a library, you can use `Expr`s to represent computations that you want to perform. This guide will walk you through how to make your own scalar UDF as an `Expr` and how to rewrite `Expr`s to inline the simple UDF.
 
+## Arrow Schema and DataFusion DFSchema
+
+Schema and DFSchema are both exist in datafusion because `Schema` provides a lightweight structure for defining data, and `DFSchema` extends it with extra information. This makes the engine could handle both simple data definitions and complex query scenarios efficiently.
```

Comment:
I would probably rephrase it saying the DFSchema is a wrapper on top of Arrow schema, extending the latter to support DF features which missing in Arrow Schema, like qualifier support, metadata, functional dependencies